### PR TITLE
Add custom ESLint rule for restricted module

### PR DESCRIPTION
**Pull request by budnix**
### Context

Since the Handsontable was modularized, there are some modules that act as [public API](https://handsontable.com/docs/8.3.0/tutorial-modules.html#modules-cheatsheet). Importing the main entry of the module can make the "dead" code elimination process more difficult (especially in older bundlers such as Webpack3 or in some cases Webpack4). So, to prevent that and to catch developer mistakes as soon as possible, the custom ESLint rule is added.

I tried using build-in [no-restricted-imports](https://eslint.org/docs/rules/no-restricted-imports#disallow-specific-imports-no-restricted-imports) rule and rules from the [eslint-plugin-import](https://www.npmjs.com/package/eslint-plugin-import) package but without success. The general problem with that rules was that it was impossible to restrict import/export modules e.q. from `../plugins` but at the same time allow import from submodules e.q. `../plugins/registry`.

I think the most promising rules are in the `eslint-plugin-import` however while playing a little bit with that I saw that the error message can not be customized. Which I think is important for this rule because the "mistake" is syntactically correct in the context of JavaScript code but in our case, it generates side-effects that require explanation.

The ESLint doesn't support adding rules or plugins inline into the config file (the plugin package has to be published). That's why after the NPM install the plugin is linked. The script is added to the `"postinstall"` hook. 

Additionally, I spotted that the plugins directory was not added to the lining process. I updated the script and fixes all issues within the files.

**NOTICE:** The PR requires installed `npm` in version 7.5.0 or higher (it's installed by default with Node v15.8.0). In the lower versions there could be an error:
image
The error could be fixed by a custom script but I think it's not necessary. The recommended Handsontable development environment is to have Node v15 and with the upcoming monorepo project, it will be required to have Node v15.

### How has this been tested?

The following pattern is considered a problem:
```js
import { getEditor } from './editors';
import { getRegisteredRendererNames } from './renderers';
import { getValidator } from './validators';
import { registerCellType } from './cellTypes';
import { registerPlugin } from './plugins';
import { getTranslatedPhrase } from './i18n';
```
The following pattern is **not** considered a problem:
```js
import { getEditor } from './editors/registry';
import { getRegisteredRendererNames } from './renderers/registry';
import { autocompleteRenderer } from './renderers/autocompleteRenderer';
import { getValidator } from './validators/registry';
import { registerCellType } from './cellTypes/registry';
import { registerPlugin } from './plugins/registry';
import { getTranslatedPhrase } from './i18n/registry';
import { warnUserAboutLanguageRegistration } from './i18n/utils';
```

The error example:
image

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1. #7473
2.
3.

### Checklist:

- [x] I have reviewed the guidelines about [Contributing to Handsontable](https://github.com/handsontable/handsontable/blob/master/CONTRIBUTING.md) and I confirm that my code follows the code style of this project.
- [x] I have signed the [Contributor License Agreement](https://docs.google.com/forms/d/e/1FAIpQLScpMq4swMelvw3-onxC8Jl29m0fVp5hpf7d1yQVklqVjGjWGA/viewform?c=0&w=1)
- [ ] My change requires a change to the documentation.

### CODE REVIEW

File: .config/plugin/eslint/rules/restricted-module-imports.js [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+const minimatch = require('minimatch');
+
+const isLocalModule = moduleName => moduleName.startsWith('.');
+
+module.exports = {
+  meta: {
+    type: 'suggestion',
+
+    docs: {
+      description: 'Allows restrict specified modules from loading by `import` or re-export',
```
Comment (nikersify) [inline .config/plugin/eslint/rules/restricted-module-imports.js:11]:
```suggestion
      description: 'Allows restricting specified modules from loading via an `import` or a re-export',
```

**Review by nikersify — APPROVED** [review]
Works great, just some optional nitpicks from me.

File: src/index.js [hunk 2]
```diff
@@ -22,6 +22,7 @@ import * as unicodeHelpers from './helpers/unicode';
 import * as domHelpers from './helpers/dom/element';
 import * as domEventHelpers from './helpers/dom/event';
 
+/* eslint-disable handsontable/restricted-module-imports */
```
[the remark this answers is not in the document]
Comment (budnix) [reply src/index.js:25]: Good point. I'll do that 👍

[the remark this answers is not in the document]
Comment (budnix) [reply src/index.js:25]: Fixed within https://github.com/handsontable/handsontable/pull/7506/commits/0624269dcc58379874164469c5dcc91d990fd603.

**Review by nikersify — APPROVED** [review]

**Pull request merged**